ucp/_libs/src/ucp_py_ucp_fxns.c

```diff
@@ -608,11 +631,43 @@ void *ucp_py_get_ep(char *ip, int listener_port)
                               send_handle);
     if (UCS_PTR_IS_ERR(request)) {
         fprintf(stderr, "unable to send UCX data message\n");
+	if (EP_CONN_FAILED == connection_status) {
+            ucp_request_cancel(ucp_py_ctx_head->ucp_worker, &request);
+            request_init(request);
+            ucp_request_free(request);
+            printf("connection failed. Retrying in %d seconds\n",
+                        ucp_py_ctx_head->connection_retry_period);
+            connection_status = EP_CONN_INIT;
+            sleep(ucp_py_ctx_head->connection_retry_period);
+            if (num_retries < EP_CONN_MAX_RETRIES) {
+                num_retries++;
+                goto retry;
```

Comment:
Whoa, I'm super-curious why you've chosen to use `goto`.  I thought that that went out of style :)

Reply:
For one, UCX makes use of it a fair bit:
```shelll
$ grep -r "goto " $HOME/ucx-github/src/ | wc -l
$ 854
```
But they're almost exclusively used for error cases unlike here. I think I'll refactor the code to use this better. Code changes coming up soon.